cylc/flow/xtrigger_mgr.py

```diff
@@ -252,50 +225,69 @@ def _get_xtrig(self, itask: TaskProxy, unsat_only: bool = False,
                 label, signature, function context, and flag for satisfied.
         """
         res = []
-        farg_templ = {
-            TMPL_TASK_CYCLE_POINT: str(itask.point),
-            TMPL_TASK_NAME: str(itask.tdef.name),
-            TMPL_TASK_IDENT: str(itask.identity)
-        }
-        farg_templ.update(self.farg_templ)
         for label, satisfied in itask.state.xtriggers.items():
             if unsat_only and satisfied:
                 continue
-            ctx = deepcopy(self.functx_map[label])
-            ctx.point = itask.point
-            kwargs = {}
-            args = []
-            # Replace legal string templates in function arg values.
-            for val in ctx.func_args:
-                try:
-                    val = val % farg_templ
-                except TypeError:
-                    pass
-                args.append(val)
-            for key, val in ctx.func_kwargs.items():
-                try:
-                    val = val % farg_templ
-                except TypeError:
-                    pass
-                kwargs[key] = val
-            ctx.func_args = args
-            ctx.func_kwargs = kwargs
-            ctx.update_command(self.suite_source_dir)
+            ctx = self.get_xtrig_ctx(itask, label)
             sig = ctx.get_signature()
             if sigs_only:
                 res.append(sig)
             else:
                 res.append((label, sig, ctx, satisfied))
         return res
 
+    # TODO TYPE HINTS
+    def get_xtrig_ctx(self, itask, label):
+        """Get a real function context from the template."""
```

Comment:
I think 

```diff
diff --git a/cylc/flow/xtrigger_mgr.py b/cylc/flow/xtrigger_mgr.py
index 4c2b3d185..67ac99fb3 100644
--- a/cylc/flow/xtrigger_mgr.py
+++ b/cylc/flow/xtrigger_mgr.py
@@ -236,9 +236,15 @@ class XtriggerManager(object):
                 res.append((label, sig, ctx, satisfied))
         return res
 
-    # TODO TYPE HINTS
-    def get_xtrig_ctx(self, itask, label):
-        """Get a real function context from the template."""
+    def get_xtrig_ctx(self, itask: TaskProxy, label: str) -> SubFuncContext:
+        """Get a real function context from the template.
+
+        Args:
+            itask (TaskProxy): task proxy
+            label (str): xtrigger label
+        Returns:
+            SubFuncContext: function context
+        """
         farg_templ = {
             TMPL_TASK_CYCLE_POINT: str(itask.point),
             TMPL_TASK_NAME: str(itask.tdef.name),

```

could work

Reply:
To address the `TODO` above the function I mean...

Reply:
But not really necessary

Reply:
Thanks, I missed my own TODO there!

Reply:
:+1: